src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -0,0 +1,389 @@
+import org.opensearch.action.admin.indices.mapping.put.PutMappingRequest;
+import org.opensearch.action.admin.indices.settings.put.UpdateSettingsRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.action.update.UpdateRequest;
+import org.opensearch.action.update.UpdateResponse;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.model.ProvisioningProgress;
+import org.opensearch.flowframework.model.State;
+import org.opensearch.flowframework.model.Template;
+import org.opensearch.flowframework.model.WorkflowState;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.opensearch.core.rest.RestStatus.INTERNAL_SERVER_ERROR;
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX;
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX_MAPPING;
+import static org.opensearch.flowframework.common.CommonValue.META;
+import static org.opensearch.flowframework.common.CommonValue.NO_SCHEMA_VERSION;
+import static org.opensearch.flowframework.common.CommonValue.SCHEMA_VERSION_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_STATE_INDEX;
+
+/**
+ * A handler for global context related operations
+ */
+public class FlowFrameworkIndicesHandler {
+    private static final Logger logger = LogManager.getLogger(FlowFrameworkIndicesHandler.class);
+    private final Client client;
+    ClusterService clusterService;
+    private static final Map<String, AtomicBoolean> indexMappingUpdated = new HashMap<>();
+    private static final Map<String, Object> indexSettings = Map.of("index.auto_expand_replicas", "0-1");
+
+    /**
+     * constructor
+     * @param client the open search client
+     * @param clusterService ClusterService
+     */
+    public FlowFrameworkIndicesHandler(Client client, ClusterService clusterService) {
+        this.client = client;
+        this.clusterService = clusterService;
+    }
+
+    static {
+        for (FlowFrameworkIndex mlIndex : FlowFrameworkIndex.values()) {
+            indexMappingUpdated.put(mlIndex.getIndexName(), new AtomicBoolean(false));
+        }
```

Comment:
Can we have this part of the constructor if it needs be created for every instance?

Reply:
This doesn't need to be created for each instance. Having a static initializers means this code only gets run when the class is first loaded. I don't need to run this code every time I create an instance of `FlowFrameworkIndicesHandler`, I only need it to be run at least once to make sure I have the updated indices in this map